frontend/packages/kubevirt-plugin/src/k8s/requests/vm/create/selectors.ts

```diff
@@ -0,0 +1,31 @@
+import { getName } from '@console/shared/src';
+import { TemplateKind } from '@console/internal/module/k8s';
+import {
+  getTemplatesOfLabelType,
+  getTemplatesWithLabels,
+} from '../../../../selectors/vm-template/advanced';
+import {
+  getFlavorLabel,
+  getOsLabel,
+  getWorkloadLabel,
+} from '../../../../selectors/vm-template/combined-dependent';
+import { TEMPLATE_TYPE_BASE } from '../../../../constants/vm';
+
+export const findTemplate = (
+  templates: TemplateKind[],
+  {
+    userTemplateName,
+    workload,
+    flavor,
+    os,
+  }: { userTemplateName: string; workload: string; flavor: string; os: string },
+): TemplateKind => {
+  if (userTemplateName) {
+    return templates.find((template) => getName(template) === userTemplateName);
+  }
+  return getTemplatesWithLabels(getTemplatesOfLabelType(templates, TEMPLATE_TYPE_BASE), [
+    getOsLabel(os),
+    getWorkloadLabel(workload),
+    getFlavorLabel(flavor),
```

Comment:
Do we need to warrry about `flavor === 'Cutome'` here ?

Reply:
this is handled by the `getFlavorLabel` function